### DISCUSSION TITLE
Producer is not valid under mingw64 on windows 10

### DISCUSSION THREAD

**Discussion by mhao1999** [mltframework/mlt#1060] [General]
I follow the build steps to generate mlt files successfully and install them in C:\Program Files (x86)\MLT, then I create an example following https://github.com/mltframework/mlt/blob/master/src/examples/play.cpp, the play.exe is created successfully, but when I check the Producer.is_valid(), it is always false.
I debug into the source code and find the mlt_repository_create return null. why?
any help will be very appreciated!
Note: I set argv[1] as /home/James/one.mp4 which is a valid mp4 file.

**Comment by bmatherly** [comment]
You are probably missing the avformat module. A good way to check is to use the melt command line application that comes with the MLT framework.

Run this:
`melt -query producers`

Do you see the avformat producer in the list?

**Reply by mhao1999** [reply]
@bmatherly great thanks! the console output is like:
# ./melt.exe -query producers
mlt_repository_init: no plugins found in "D:\mingw\player\build\Desktop_Qt_5_15_15_shared_MinGW_w64_
MINGW64_MSYS2-Debug\lib\mlt"
---
producers:
...

no producer? how to install such plugins, I already use pacman -S to install the latest ffmpeg into c:\msys64\mingw64\bin, and I already copy all dlls in such folder to the melt.exe folder.

**Reply by mhao1999** [reply]
this is what ffmpeg printing:
# ffmpeg
ffmpeg version 7.1 Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 14.2.0 (Rev2, Built by MSYS2 project)
  configuration: --prefix=/mingw64 --target-os=mingw32 --arch=x86_64 --cc=gcc --cxx=g++ --disable-de
bug --disable-stripping --disable-doc --enable-dxva2 --enable-d3d11va --enable-d3d12va --enable-frei
0r --enable-gmp --enable-gnutls --enable-gpl --enable-iconv --enable-libaom --enable-libass --enable
-libbluray --enable-libcaca --enable-libdav1d --enable-libfontconfig --enable-libfreetype --enable-l
ibfribidi --enable-libgme --enable-libgsm --enable-libharfbuzz --enable-libjxl --enable-libmodplug -
-enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enab
le-libopus --enable-libplacebo --enable-librsvg --enable-librtmp --enable-libssh --enable-libsoxr --
enable-libspeex --enable-libsrt --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-l
ibx264 --enable-libx265 --enable-libxvid --enable-libvpx --enable-libwebp --enable-libxml2 --enable-
libzimg --enable-libzvbi --enable-openal --enable-pic --enable-postproc --enable-runtime-cpudetect -
-enable-swresample --enable-version3 --enable-vulkan --enable-zlib --enable-librav1e --enable-libvpl
 --enable-libsvtav1 --enable-liblc3 --enable-amf --enable-nvenc --logfile=config.log --enable-shared
  libavutil      59. 39.100 / 59. 39.100
  libavcodec     61. 19.100 / 61. 19.100
  libavformat    61.  7.100 / 61.  7.100
  libavdevice    61.  3.100 / 61.  3.100
  libavfilter    10.  4.100 / 10.  4.100
  libswscale      8.  3.100 /  8.  3.100
  libswresample   5.  3.100 /  5.  3.100
  libpostproc    58.  3.100 / 58.  3.100
Universal media converter
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

**Comment by bmatherly** [comment]
You should read the output of the MLT configure and build steps to see if the modes that you need are being compiled.

**Reply by mhao1999** [reply]
ok, thanks, the demo can run now, I think the cause is I install the output files to wrong place.
